mountpoint-s3/tests/fuse_tests/write_test.rs

```diff
@@ -778,21 +922,47 @@ where
     let mut hello_contents = String::new();
     read_fh.read_to_string(&mut hello_contents).unwrap();
     assert!(hello_contents.is_empty());
-    drop(read_fh);
+}
 
-    // Open with O_WRONLY and write something to the file
-    let mut options = File::options();
-    let mut write_fh = options.write(true).truncate(true).open(&path).unwrap();
-    write_fh.write_all(b"second overwrite").expect("write should succeed");
-    write_fh.sync_all().unwrap();
-    drop(write_fh);
+#[cfg(feature = "s3_tests")]
+#[test_case(true; "write_only")]
+#[test_case(false; "read_write")]
+fn overwrite_truncate_test_s3(write_only: bool) {
+    overwrite_truncate_test(fuse::s3_session::new, "overwrite_truncate_test", write_only);
+}
 
-    // Check the new file content
+#[test_case(true; "write_only")]
+#[test_case(false; "read_write")]
+fn overwrite_truncate_test_mock(write_only: bool) {
+    overwrite_truncate_test(fuse::mock_session::new, "overwrite_truncate_test", write_only);
+}
+
+fn read_overwrite_read_test<F>(creator_fn: F, prefix: &str)
```

Comment:
Similarly, do we expect to see bad FD if the read occurs too soon after the write or write too soon after read?

Reply:
This is currently passing on `main`, we'll review again when changing behavior.

Reply:
read after write, yes. write after read is safe on `main` (may not be if we merge #751)